### PR TITLE
Storyai 2 db schema

### PR DESCRIPTION
**Pull request by Donnadieu**

### CODE REVIEW

File: app/models/user_story.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class UserStory < ApplicationRecord
+  belongs_to :user
+  belongs_to :story, required: false
+
+  validates :user_id, presence: true
+  validates :user_id, uniqueness: { scope: :story_id }
```
Comment (uchilaka) [inline app/models/user_story.rb:6]: Does this mean each story can only belong to 1 user?

File: config/routes.rb [hunk 2]
```diff
@@ -21,6 +21,9 @@
         end
       end
     end
+    namespace :v1 do
+      resources :stories, only: [:create, :update, :index, :show, :destroy]
```
Comment (uchilaka) [inline config/routes.rb:25]: I think we definitely want to do a soft destroy, to either:
- Exclude the stories that are "deleted" from showing up for the customer 
- Mark the story for deletion and go through our own back office processing for deleting stories

Either way, we want to include in our terms that the data is at our discretion, with the exception of customers making a request to have their data deleted, which then we will comply with within 30 days or something like that

Comment (Donnadieu) [reply config/routes.rb:25]: Agreed. Destroy currently is not implemented in the controller. I don't think should be mvp

Comment (uchilaka) [reply config/routes.rb:25]: The data is kind of our only asset to start with (per our chat from earlier)
- To train our own LLM
- To monetize for publication

File: app/controllers/api/v1/stories_controller.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    if @story.update(story_params)
+      render json: @story
+    else
+      render json: @story.errors, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    # Rerturn not implemented
+    head :not_implemented
```
Comment (uchilaka) [inline app/controllers/api/v1/stories_controller.rb:40]: Saw this after my comments about soft deletion 😁

File: config/routes.rb (hunk 2 above)
Comment (Donnadieu) [reply config/routes.rb:25]: Definitely. I envision an `status` column maybe managed by a state machine

File: spec/models/user_story_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  context 'uniqueness' do
+    it 'validates uniqueness of :user_id scoped to :story_id' do
+      # Create a valid user_story record with a user and story
+      user = FactoryBot.create(:user)
+      story = FactoryBot.create(:story)
+      user_story = FactoryBot.create(:user_story, user: user, story: story)
+
+      # Build a new user_story with the same user and story
+      new_user_story = FactoryBot.build(:user_story, user: user, story: story)
```
Comment (uchilaka) [inline spec/models/user_story_spec.rb:22]: 👍🏾

**Review by uchilaka — APPROVED** [review]
LGTM!

**Pull request merged**